### PR TITLE
update PULL_REQUEST_TEMPLATE and CONTRIBUTING

### PR DESCRIPTION
**Pull request by ebuchman**
Ref #3654

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ Thank you for considering making contributions to Tendermint and related reposit
 
 Please follow standard github best practices: fork the repo, branch from the tip of develop, make some commits, and submit a pull request to develop. See the [open issues](https://github.com/tendermint/tendermint/issues) for things we need help with!
 
+Before making a pull request, please open an issue describing the
+change you would like to make. If an issue for your change already exists,
+please comment on it that you will submit a pull request. Be sure to reference the issue in the opening
```
Comment (melekes) [inline CONTRIBUTING.md]: `(Refs #XXXX)`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,6 +4,12 @@ Thank you for considering making contributions to Tendermint and related reposit
...
+comment of your pull request. If your change is substantial, you will be asked
+to write a more detailed design document before submitting code changes.
```
Comment (melekes) [inline CONTRIBUTING.md]: ADR example?

File: CONTRIBUTING.md (hunk 1 above)
Comment (Liamsi) [reply CONTRIBUTING.md]: We could make this a placeholder in the template maybe?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
-<!-- Thanks for filing a PR! Before hitting the button, please check the following items.-->
+<!--
 
+Thanks for filing a PR! Before hitting the button, please check the following items.
+Please note that every non-trivial PR must reference an issue that explains the
+changes in the PR.
+
+-->
+
+* [ ] Referenced an issue explaining the change
```
Comment (thanethomson) [inline .github/PULL_REQUEST_TEMPLATE.md]: The issue highlights the need, right?

```suggestion
* [ ] Referenced an issue explaining the need for the change
```

**Pull request merged**